zend_abstract_interface/zai_string/string.c

```diff
@@ -0,0 +1,7 @@
+#include "string.h"
+
+char *ZAI_STRING_EMPTY_PTR = "";
```

Comment:
What's the goal of this variable? Compared to just `""` at usage sites?

Reply:
Is this a hack to avoid allocating and freeing an empty string?
I don't like this as it doesn't allow for a trivial realloc(), but one does have to check whether it's ZAI_STRING_EMPTY_PTR.

Reply:
Avoiding an allocation for an empty string is better than having an extra check in things like realloc, which notably we do not even use at the moment in `zai_string`.

Reply:
`""` would be a string literal even without a variable. Also the compiler would optimize the storage to make sure you have a single string literal even if you have several places where you use `""`.
Or is this to compare the address of the variable ? I might be missing something, I have not checked usage yet.

Reply:
To be clear I don't mind both, I just did not see where the allocation would be happening.

Reply:
@r1viollet This is to compare the address.